storm-core/src/jvm/org/apache/storm/stats/CommonStats.java

```diff
@@ -23,10 +23,11 @@
 import org.apache.storm.metric.api.IMetric;
 import org.apache.storm.metric.internal.MultiCountStatAndMetric;
 import org.apache.storm.metric.internal.MultiLatencyStatAndMetric;
+import org.apache.storm.utils.Utils;
 
 @SuppressWarnings("unchecked")
 public abstract class CommonStats {
-    public static final int NUM_STAT_BUCKETS = 20;
+    public static final int NUM_STAT_BUCKETS =(int) Utils.readStormConfig().get("num.stat.buckets");
```

Comment:
Could we find other approach, like passing that value from caller of CommonStats?
I think we're reading storm config file again and again which looks redundant, and even doing it for constant doesn't look good.